### PR TITLE
Remove Streamable

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

`Streamable` has been a legacy artifact from pre-Java8 times,
`Writeable` is the successor.

This removes all remaining usages we had.

There is probably more potential in making attributes final, I went
through things with a "don't spend too much time on making more changes
than necessary" approach. We can tweak things further from here.

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: blob/src/main/java/io/crate/blob/PutChunkResponse.java [hunk 1]
```diff
@@ -21,8 +21,16 @@
...
 package io.crate.blob;
 
+import org.elasticsearch.common.io.stream.StreamInput;
+
+import java.io.IOException;
+
 public class PutChunkResponse extends BlobTransferResponse {
 
     public PutChunkResponse() {
     }
```
[the remark this answers is not in the document]
Comment (mfussenegger) [reply blob/src/main/java/io/crate/blob/PutChunkResponse.java]: The parent constructors are empty, so it doesn't change behavior. 

Can add it for consistency if you'd like.

**Comment by mfussenegger** [comment]
> Does the first commit compile and work standalone?

yep it should, I'll double check to make sure it does.

File: blob/src/main/java/io/crate/blob/PutChunkResponse.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (seut) [reply blob/src/main/java/io/crate/blob/PutChunkResponse.java]: just thought of consistency as this pr also added some super calls to empty constructors. also fine for me to leave it as it is.

[the remark this answers is not in the document]
Comment (mfussenegger) [reply blob/src/main/java/io/crate/blob/PutChunkResponse.java]: :+1:  will follow up on this (maybe we can remove the empty constructor altogether)

**Pull request merged**